### PR TITLE
build: Faster `make start`, faster smoke jobs

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,6 +51,7 @@ CONTROLLER_IMAGE_FILE  := dist/controller-image.marker
 STATIC_BUILD          ?= true
 # should we build the static files?
 STATIC_FILES          ?= $(shell [ $(DEV_BRANCH) = true ] && echo false || echo true)
+START_UI              ?= $(shell [ "$(CI)" != "" ] && echo true || echo false)
```
Comment (alexec) [inline Makefile:54]: we were building and running the UI for smoke/e2e tests on CI

File: test/e2e/manifests/mysql/kustomization.yaml [hunk 2]
```diff
@@ -2,8 +2,10 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
 resources:
-  - ../../../../manifests/quick-start/mysql
-  - https://github.com/argoproj/argo-events/manifests/base/crds?ref=stable
+- ../../../../manifests/quick-start/mysql
+- https://raw.githubusercontent.com/argoproj/argo-events/stable/manifests/base/crds/argoproj.io_eventbus.yaml
```
Comment (alexec) [inline test/e2e/manifests/mysql/kustomization.yaml:6]: missed this change in my last PR

File: ui/src/app/webpack.config.js [hunk 3]
```diff
@@ -79,6 +79,9 @@ const config = {
     historyApiFallback: {
       disableDotRule: true
     },
+    watchOptions: {
```
Comment (alexec) [inline ui/src/app/webpack.config.js:82]: this stops my fans from running 100%

File: .golangci.yml [hunk 4]
```diff
@@ -7,8 +7,11 @@ run:
   skip-files:
     - server/static/files.go
   build-tags:
-    - e2e
```
Comment (alexec) [inline .golangci.yml:10]: split the e2e tests up by tag to make it easier to run groups of them by tag

**Review by simster7 — APPROVED** [review]

**Pull request merged**